### ISSUE TITLE
typst table overflowing with gt and row names

### ISSUE THREAD

**Issue by olivroy** [quarto-dev/quarto-cli#10228]
### Bug description

A table doesn't wrap its output

### Steps to reproduce

````qmd
---
title: "Sortable Tables Test"
format: typst
editor: source
---


```{r}
mtcars |> head() |> gt::gt(rownames_to_stub = TRUE)
```

````

### Expected behavior

I would expect the text to be wrapped.

### Actual behavior

The text overflows (see Hornet Sport).
![image](https://github.com/quarto-dev/quarto-cli/assets/52606734/2dfe3b0d-be30-457e-9dd3-1e34d3d48de5)

### Your environment

Windows, RStudio 2024.07.0

### Quarto check output

```bash
Quarto 1.5.53
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.1: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.49
      Path: C:\Users\name\AppData\Local\Programs\RStudio\resources\app\bin\quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\name\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.12.4
      Path: C:/Users/name/AppData/Local/Programs/Python/Python312/python.exe
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with py -m pip install jupyter

[>] Checking R installation...........OK
      Version: 4.3.2
      Path: C:/PROGRA~1/R/R-43~1.2
      LibPaths:
        - C:/Users/name/AppData/Local/R/win-library/4.3
        - C:/Program Files/R/R-4.3.2/library
      knitr: 1.47
      rmarkdown: 2.27

[>] Checking Knitr engine render......OK
```

Mentioned in https://github.com/rstudio/gt/discussions/1584

**Comment by gordonwoodhull** [comment]
This is a Typst issue, unrelated to Quarto or gt. Typst does not deal very well with a table that is too wide for the page.

I ran into it frequently while testing the Typst CSS feature. The only workarounds I found were reducing the number of columns or reducing the font size.

In this case, it looks like the word "Sportabout" is perhaps too wide for the space Typst has allocated for the column. It does not know how to wrap this, so it gives up. (I am speculating here.)

When Quarto encounters an HTML table that is too wide for the page, it adds horizontal scrolling with `overflow: x`. Being a print format, Typst does not have that escape hatch.
